### PR TITLE
fix: resolve symbol-upload outside the package folder

### LINKED ISSUE #150: G1: Symbol upload silently fails for registry and git package installs
## Problem
`UploadSymbols` resolves `Packages/com.bugsplat.unity/Editor/<exe>` as a physical path (`Editor/PostBuild.cs`) — nonexistent for PackageCache installs. The fallback downloads the uploader *into the package folder*, which is immutable for registry installs; `Process.Start` then throws uncaught and **the build still succeeds with no symbols**. Most users install exactly this way (git URL / OpenUPM), so the flagship Windows-native story ships with broken symbol upload for the majority path.

## Fix
Resolve via `PackageInfo.FindForAssembly` (`PostProcessWindows` already does this correctly), download to `Temp/`, surface non-zero exit codes, and add an optional fail-the-build toggle. Add editor-mode tests for the path resolution (#132 J3 notes the Editor surface has zero coverage).

Sliced from the 5.0.0 audit #132; full context and benchmarks there.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by bobbyg603**
Closes #150

## What was broken

`UploadSymbols` resolved the uploader at `Path.GetFullPath("Packages/com.bugsplat.unity/Editor/<exe>")`. That path only physically exists for embedded/local installs — registry and git URL installs resolve under `Library/PackageCache/com.bugsplat.unity@<hash>/`, so the file was never found there.

The uploader binaries are gitignored and never ship in the package, so the download fallback ran on *every* install, not just the broken ones. For PackageCache installs it tried to write into a directory that does not exist (and that Unity owns and may re-extract at any time). `WebClient.DownloadFile` threw, the error was logged and swallowed, and `Process.Start` then threw uncaught with a `FileName` pointing at a nonexistent binary — leaving a build that reported success with no symbols uploaded.

This affects the majority install path (git URL / OpenUPM), on every platform that goes through `UploadSymbols` (Windows, macOS, Android).

## What changed

All of it in `Editor/PostBuild.cs`, confined to the uploader path resolution and download:

- New `GetSymUploaderPath()` resolves the package root via `UnityEditor.PackageManager.PackageInfo.FindForAssembly`, the same way `PostProcessWindows` already does, with the old `Packages/com.bugsplat.unity` path kept only as the fallback when `FindForAssembly` returns null. An uploader already sitting in the package's `Editor/` folder is used in place; otherwise the path resolves to `Temp/`, which is writable for every install type, is not scanned by the asset importer (no `.meta` churn), and never mutates the package.
- `DownloadSymbolUpload` now creates the destination directory and returns whether it succeeded. A failed download or a failed `chmod +x` short-circuits to `onCompleted(-1)` instead of falling through to a doomed `Process.Start`.
- `Process.Start` is wrapped in a try/catch, so a launch failure is logged and surfaced as a non-zero exit code to the existing callbacks rather than escaping the `[PostProcessBuild]` callback.

Deliberately not touched, to keep this mergeable alongside #151 and #162: the `#elif UNITY_EDITOR_WIN` dispatch in `OnPostprocessBuild` and the Android upload callback.

## Verified

- `dotnet build` of the full `Runtime/` + `Editor/` source set against Unity 6000.5.6f1's managed assemblies, in three define configurations: `UNITY_EDITOR`, `UNITY_EDITOR;UNITY_EDITOR_WIN` (with `UnityEditor.WindowsStandalone.Extensions.dll`), and `UNITY_EDITOR;UNITY_ANDROID` (with `UnityEditor.Android.Extensions.dll` + `Unity.Android.Types.dll`). 0 errors in all three; the only warnings are pre-existing (`_platform` unused, two never-assigned fields in `BugSplat.cs`).
- Confirmed the bug is real in current `main` rather than drifted audit notes: the hardcoded path was still at `PostBuild.cs:545`, and `.gitignore` confirms the uploader binaries are never committed, so the download path is always exercised.

## Not verified

- **No Unity run of any kind.** The compile check above is `dotnet build` against Unity's DLLs, not a Unity build, so nothing here was exercised at runtime: no actual player build, no real download to `Temp/`, no real upload to BugSplat, and no confirmation from an actual PackageCache install.
- The `#if UNITY_IOS` block was not compiled (needs the iOS Xcode extension assemblies). It does not call `UploadSymbols` and is unchanged by this PR.
- `Temp/` is cleared when the editor quits, so the uploader is re-downloaded once per editor session. That is the tradeoff the issue asked for and it is not benchmarked here.
- **No tests were added.** The issue also asks for editor-mode tests for path resolution; the repo has only a `BugSplat.Unity.RuntimeTests` assembly, so that needs a new Editor test assembly and is left as follow-up under #132 (J3). The optional fail-the-build toggle from the issue is also not implemented — it needs a new serialized field on `BugSplatOptions` plus editor UI, which would collide with the other in-flight `PostBuild.cs` PRs.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by daveplunkett — APPROVED** [review]

File: Editor/PostBuild.cs [hunk 1]
```diff
@@ -582,13 +608,15 @@ private static void UploadSymbols(string artifactsDirPath, string globPattern, B
 		onCompleted(uploadSymProcess.ExitCode);
 	}
 
-	private static void DownloadSymbolUpload(string destinationPath)
+	private static bool DownloadSymbolUpload(string destinationPath)
 	{
 		var varient = Path.GetFileName(destinationPath);
 		var fileUrl = $"https://app.bugsplat.com/download/{varient}";
```
[the remark this answers is not in the document]
Comment (bobbyg603) [reply Editor/PostBuild.cs]: Fixed in 1fd6cc1 — renamed `varient` → `variant` everywhere in `DownloadSymbolUpload` (declaration, URL, and the three log lines). Note the interpolations only ever emitted the variable's *value* (the uploader filename), so no logs/URLs actually carried the misspelling — it was confined to source.

File: Editor/PostBuild.cs [hunk 2]
```diff
@@ -568,7 +583,18 @@ private static void UploadSymbols(string artifactsDirPath, string globPattern, B
...
+		Process uploadSymProcess;
+		try
+		{
+			uploadSymProcess = Process.Start(symUploadProcessInfo);
+		}
+		catch (Exception ex)
+		{
+			Debug.LogError($"BugSplat. Failed to start {symbolUploadPath}. Error: {ex.Message}");
+			onCompleted(-1);
+			return;
```
[the remark this answers is not in the document]
Comment (bobbyg603) [reply Editor/PostBuild.cs]: Fixed in 1fd6cc1 — the `Process.Start` catch, the chmod catch (your line-670 sibling), and the unexpected-download catch now interpolate `{ex}` (full type + stack) instead of `{ex.Message}`. Intentionally left the `WebException` catch on `ex.Message`: it's the expected network-failure path, the message is self-describing (status/timeout), and the URL is already in the log line.

**Comment by bobbyg603** [comment]
Also addressed the suppressed comment from the Copilot review: the chmod block's `PostBuild:` log prefix is now `BugSplat.` to match the rest of the file, and its catch logs the full exception (1fd6cc1). Housekeeping note: 1fd6cc1 accidentally flipped the file to LF line endings; 1d541bb restores CRLF, so those two commits show whole-file churn but the net PR diff is just the intended lines.
